src/calculation/HTMLBuilder.ts

```diff
@@ -139,6 +145,40 @@ export function generateHTML(
   return result;
 }
 
+/**
+ * Destroys the input annotations and returns the sorted annotations in a logical order:
+ * First, population expressions, according to the order set in the measure
+ * Second, non-function expressions in alphabetical order
+ * Third, function expressions in alphabetical order
+ */
+export function sortAnnotations(measure: fhir4.Measure, groupId: string, annotations: StatementAnnotation[]) {
```

Comment:
I personally think we can simply this code a lot by changing the approach. I think we are doing a lot of unnecessary array manipulation and memory allocation for this problem. My suggestion is to consider sorting the statementResults in the original function, before they are converted into annotations, and avoid any type of "removal" during the re-ordering. This should be achievable with just one `.sort`. Here's some pseudocode:

```
  relevantStatements.sort((s1, s2) => {
    if (one of them is a function) {
        // ensure the one that is the function comes second
    }
    
    if (s1 is a population expression) return -1;
    if (s2 is a population expression) return 1;

    return s1.statementName <= s2.statementName ? -1 : 1;
  });
```

A couple things to note:

1. Each statementResult has a handy `.isFunction` attribute
2. For the detection of populationExpressions, this function does a .findIndex on all the annotations within each measure population. The efficiency could be improved here by using a `Set` and assembling a unique set of strings that is all of the population expressions used in the measure group. Then we can do a `O(1)` lookup within the `sort` that I outlined to check the `if(__ is a population expression)`

I implemented a version of the above pseudocode locally and got the same results as this implementation. (sorry)

Reply:
So one thing I was trying to do was to ensure that the population expressions were in the same order expressed by the measure. Do you think that's valuable, or we just don't care as long as they're clustered at the top?